### DISCUSSION TITLE
Active state on links lost while loader is resolving

### DISCUSSION THREAD

**Discussion by damianstasik** [TanStack/router#2076] [Q&A]
I'd like to ask whether the behavior I've encountered is expected or if it might be a bug. Here is a simple reproduction of the behavior:
https://stackblitz.com/edit/vitejs-vite-kmsvy5?file=src%252Fmain.tsx

https://github.com/user-attachments/assets/47278404-f30d-49bb-b243-0224334a57d8

In short: when you navigate through the pages at the bottom, the languages above lose their active state while the loader awaits the promise. This is unexpected for me, as I would expect the active state to remain on the "English" link since the language did not change. Is this a bug? If not, is there any way to opt out of this behavior?

**Comment by damianstasik** [comment]
After upgrading from `1.46.3` to `1.46.8` the active state is no longer being applied to the English language link once you move to Contact or About, it's only activated when you go back to Home. Is there a way to create a link that points to a specific search query param, but without locking it to particular route params?

**Reply by schiller-manuel** [reply]
it's probably best to create a github issue instead of using discussions since those are not monitored as closely
